### PR TITLE
Update structured-merge-diff to latest version

### PR DESCRIPTION
**Pull request by apelisse**
Update structured-merge-diff repository since it's been optimized quite significantly.

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -473,7 +473,7 @@ replace (
 	modernc.org/strutil => modernc.org/strutil v1.0.0
 	modernc.org/xc => modernc.org/xc v1.0.0
 	sigs.k8s.io/kustomize => sigs.k8s.io/kustomize v2.0.3+incompatible
-	sigs.k8s.io/structured-merge-diff => sigs.k8s.io/structured-merge-diff v0.0.0-20190724202554-0c1d754dd648
+	sigs.k8s.io/structured-merge-diff => sigs.k8s.io/structured-merge-diff v0.0.0-20190724202554-6149e4549fc
```
Comment (cblecker) [inline go.mod]:
```suggestion
	sigs.k8s.io/structured-merge-diff => sigs.k8s.io/structured-merge-diff v0.0.0-20190817042607-6149e4549fca
```

Comment (apelisse) [reply go.mod]: Thank you, that's not very obvious ;-)

Comment (apelisse) [reply go.mod]: Same error, still not sure what to do

**Pull request merged**